hooks/useFetch.js

```diff
@@ -141,30 +146,26 @@ const useFetchMany = (urls) => {
   }, [key])
 
   useEffect(() => {
-    const base = from(urls).pipe(
-      mergeMap((url) => from(Object.entries(fetchData(url)))),
-    )
-
-    const errors = new Subject().pipe(
-      filter(isError),
-      pluck('error'),
-      updateBy('url'),
-    )
+    // only take one for no auto-refresh
+    let base = from(urls).pipe(map((url) => fetchData(url).pipe(take(1))))
+    if (refresh) {
+      base = from(urls).pipe(map((url) => fetchData(url)))
+    }
 
-    const sub = from(base)
+    const sub = base
       .pipe(
-        filter(([stream]) => stream === 'data'),
-        pluck(1),
-        mergeAll(),
-        tap(errors),
-        // This is dumb and AFAICT the most elegant way to silently drop
-        // errors.
-        filter((i) => !isError(i)),
-        dematerialize(),
-        updateBy('url'),
+        combineAll(),
```

Comment:
Does this end up waiting for all to complete? The main reason I did this implementation was to allow results to stream back (especially when there's partial errors). I'd like to get results from some of the endpoints even when there's issues across them all (imagine a timeout holding everything up for 10 seconds).

Reply:
Yes, it will lump the individual streams together and emit them as one value. You're right, this will result in the joint completion taking as long as the slowest one to complete. Let me improve on that.

I generally wouldn't recommend lumping unrelated requests together unless the consumer(s) requires it. I mean, I suppose `useFetchMany` is more of a convenience API.